src/api/auth/client-cert.strategy.ts

```diff
@@ -0,0 +1,24 @@
+import { PeerCertificate, Strategy } from 'passport-client-cert'
+import { PassportStrategy } from '@nestjs/passport'
+import { Injectable, UnauthorizedException } from '@nestjs/common'
+
+@Injectable()
+export class ClientCertStrategy extends PassportStrategy(Strategy) {
+  private whitelistedCns: string[] = []
+
+  constructor() {
+    super()
+    // TO DO : Refacto du CC dans le ||
+    this.whitelistedCns.push(process.env.WHITELIST_CN || 'CC')
+  }
+
+  async validate(clientCert: PeerCertificate): Promise<any> {
+    const cn = clientCert && clientCert.subject && clientCert.subject.CN
+    if (!this.whitelistedCns.includes(cn)) {
+      console.error('Unauthorized: Client cert cn : %s is not whitelisted', cn)
+      throw new UnauthorizedException()
+    }
+    console.log(cn, 'Client Authorized')
```

Comment:
Log à remettre en forme

Reply:
Est-ce que ça pourrait être : 
   Error :    this.logger.error(`[AUTH] Unauthorized: Client cert cn: ${cn} is not whitelisted`)

 OK:    this.logger.log(`[AUTH] Client Authorized: ${cn}`)